components/Article/Codebox/__tests__/index.test.tsx

```diff
@@ -0,0 +1,121 @@
+import { fireEvent, render, screen } from '@testing-library/react';
+import { highlight, languages } from 'prismjs';
+import { sanitize } from 'isomorphic-dompurify';
+import { IntlProvider } from 'react-intl';
+
+import Codebox, { replaceLabelLanguages, replaceLanguages } from '../index';
+
+const navigatorClipboardWriteTextSpy = jest
+  .fn()
+  .mockImplementation(() => Promise.resolve());
+
+Object.defineProperty(window.navigator, 'clipboard', {
+  writable: true,
+  value: {
+    writeText: navigatorClipboardWriteTextSpy,
+  },
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('Replacer tests', (): void => {
+  it('replaceLabelLanguages', (): void => {
+    expect(replaceLabelLanguages('language-console')).toBe('language-bash');
+  });
+
+  it('replaceLanguages', (): void => {
+    expect(replaceLanguages('language-mjs')).toBe('language-js');
+    expect(replaceLanguages('language-cjs')).toBe('language-js');
+    expect(replaceLanguages('language-javascript')).toBe('language-js');
+    expect(replaceLanguages('language-console')).toBe('language-bash');
+    expect(replaceLanguages('language-shell')).toBe('language-bash');
+  });
+});
+
+describe('Codebox component (one lang)', (): void => {
+  const code = 'const a = 1;';
+  const textToCopy = sanitize(highlight(code, languages.js, 'js'));
+
+  it('renders correctly', (): void => {
+    const { container } = render(
+      <IntlProvider locale="en" onError={() => {}}>
+        <Codebox>
+          <code className="language-js">{textToCopy}</code>
+        </Codebox>
+      </IntlProvider>
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('should copy content', async () => {
+    render(
+      <IntlProvider locale="en" onError={() => {}}>
+        <Codebox>
+          <code className="language-js">{textToCopy}</code>
+        </Codebox>
+      </IntlProvider>
+    );
+
+    const buttonElement = screen.getByText('components.codeBox.copy');
+    fireEvent.click(buttonElement);
+
+    expect(navigatorClipboardWriteTextSpy).toHaveBeenCalledTimes(1);
+    expect(navigatorClipboardWriteTextSpy).toHaveBeenCalledWith(
+      textToCopy.toString()
+    );
+  });
+});
+
+describe('Codebox component (multiple langs)', (): void => {
+  const code = `const http = require('http');
+  -------
+  import http from 'http';`;
+
+  it('renders correctly', (): void => {
+    const { container } = render(
+      <IntlProvider locale="en" onError={() => {}}>
+        <Codebox>
+          <code className="language-js|language-js">{code}</code>
+        </Codebox>
+      </IntlProvider>
+    );
+    expect(container).toMatchSnapshot();
+  });
+
+  it('switch between languages', async () => {
+    render(
+      <IntlProvider locale="en" onError={() => {}}>
+        <Codebox>
+          <code className="language-cjs|language-mjs">{code}</code>
+        </Codebox>
+      </IntlProvider>
+    );
+
+    const buttonElement = screen.getByText('cjs');
+    userEvent.click(buttonElement);
```

Comment:
This should be async with `const user = userEvent.setup()`

Reply:
`userEvent.setup` is conflicting our `navigator` mocks
```
  ● Codebox component (multiple langs) › switch between languages

    TypeError: Cannot redefine property: clipboard
        at Function.defineProperty (<anonymous>)
```

Reply:
Then we don't need a mock if we do userEvent.setup(). And probably there's an API from testing library to test this navigator API.